src/more_ops.rs

```diff
@@ -18,56 +18,56 @@ use crate::serialize::node_to_bytes;
 
 use sha2::{Digest, Sha256};
 
-const ARITH_BASE_COST: Cost = 4;
-const ARITH_COST_PER_ARG: Cost = 8;
-const ARITH_COST_PER_LIMB_DIVIDER: Cost = 64;
+const ARITH_BASE_COST: Cost = 1;
+const ARITH_COST_PER_ARG: Cost = 3;
+const ARITH_COST_PER_LIMB_DIVIDER: Cost = 32;
 
-const LOG_BASE_COST: Cost = 6;
-const LOG_COST_PER_ARG: Cost = 8;
-const LOG_COST_PER_LIMB_DIVIDER: Cost = 64;
+const LOG_BASE_COST: Cost = 1;
+const LOG_COST_PER_ARG: Cost = 3;
+const LOG_COST_PER_LIMB_DIVIDER: Cost = 32;
 
-const LOGNOT_BASE_COST: Cost = 12;
-const LOGNOT_COST_PER_BYTE_DIVIDER: Cost = 512;
+const LOGNOT_BASE_COST: Cost = 3;
+const LOGNOT_COST_PER_BYTE_DIVIDER: Cost = 32;
 
-const MUL_BASE_COST: Cost = 2;
-const MUL_COST_PER_OP: Cost = 18;
-const MUL_LINEAR_COST_PER_BYTE_DIVIDER: Cost = 64;
-const MUL_SQUARE_COST_PER_BYTE_DIVIDER: Cost = 44500;
+const MUL_BASE_COST: Cost = 1;
+const MUL_COST_PER_OP: Cost = 8;
+const MUL_LINEAR_COST_PER_BYTE_DIVIDER: Cost = 16;
+const MUL_SQUARE_COST_PER_BYTE_DIVIDER: Cost = 16384;
 
-const GR_BASE_COST: Cost = 19;
-const GR_COST_PER_LIMB_DIVIDER: Cost = 64;
+const GR_BASE_COST: Cost = 5;
+const GR_COST_PER_LIMB_DIVIDER: Cost = 32;
 
-const CMP_BASE_COST: Cost = 16;
+const CMP_BASE_COST: Cost = 2;
 const CMP_COST_PER_LIMB_DIVIDER: Cost = 64;
 
-const STRLEN_BASE_COST: Cost = 18;
-const STRLEN_COST_PER_BYTE_DIVIDER: Cost = 4096;
+const STRLEN_BASE_COST: Cost = 2;
+const STRLEN_COST_PER_BYTE_DIVIDER: Cost = 128;
 
-const CONCAT_BASE_COST: Cost = 4;
-const CONCAT_COST_PER_ARG: Cost = 8;
-const CONCAT_COST_PER_BYTE_DIVIDER: Cost = 830;
+const CONCAT_BASE_COST: Cost = 1;
+const CONCAT_COST_PER_ARG: Cost = 1;
+const CONCAT_COST_PER_BYTE_DIVIDER: Cost = 32;
 
-const DIVMOD_BASE_COST: Cost = 29;
-const DIVMOD_COST_PER_LIMB_DIVIDER: Cost = 64;
+const DIVMOD_BASE_COST: Cost = 11;
+const DIVMOD_COST_PER_LIMB_DIVIDER: Cost = 16;
 
-const DIV_BASE_COST: Cost = 29;
-const DIV_COST_PER_LIMB_DIVIDER: Cost = 64;
+const DIV_BASE_COST: Cost = 9;
+const DIV_COST_PER_LIMB_DIVIDER: Cost = 32;
 
-const SHA256_BASE_COST: Cost = 3;
-const SHA256_COST_PER_ARG: Cost = 8;
+const SHA256_BASE_COST: Cost = 2;
+const SHA256_COST_PER_ARG: Cost = 1;
 const SHA256_COST_PER_BYTE_DIVIDER: Cost = 64;
 
-const SHIFT_BASE_COST: Cost = 21;
-const SHIFT_COST_PER_BYTE_DIVIDER: Cost = 256;
+const SHIFT_BASE_COST: Cost = 4;
+const SHIFT_COST_PER_BYTE_DIVIDER: Cost = 32;
 
 const BOOL_BASE_COST: Cost = 1;
-const BOOL_COST_PER_ARG: Cost = 8;
+const BOOL_COST_PER_ARG: Cost = 3;
 
-const POINT_ADD_BASE_COST: Cost = 213;
-const POINT_ADD_COST_PER_ARG: Cost = 358;
+const POINT_ADD_BASE_COST: Cost = 316;
+const POINT_ADD_COST_PER_ARG: Cost = 4200;
```

Comment:
These increases must be because we are using the Rust library, instead of our own Relic-based BLS-128.

Reply:
but it's a good point, long term we might want to use the same bls implementation

Reply:
Can you explain what these operations are? Which one is used for the AGG_SIG condition (which requires verifying a pairing operation). 

From what I've seen in our bls library, verifying a single pairing takes 3ms, but when batch verifying it takes 1ms (so we should use the batch verification cost). Does this kind-of align with the costs in this file?

Reply:
Although clvm generates `(AGG_SIG ...)` and `(AGG_SIG_ME ...)` conditions as output, the actual pairing is deferred to the condition checker, which is written in python, and adds costs proportionate to the number of these conditions. So none of these costs in this PR correspond to pairing.

Instead, look here: https://github.com/Chia-Network/chia-blockchain/blob/main/src/consensus/cost_calculator.py#L38

Reply:
Probably @arvidn should benchmark the ops here to make sure their costs are normalized to the same scale as his other benchmarks.

Reply:
the new costs represents approximately 0.1 microsecond of run-time per 1 cost, on my MacBook Pro.
If the `AGG_SIG` operation takes 1 ms (amortized) on my MacBook, it would correspond to roughly 10000 in cost.

However, looking at the link richard posted, it looks like we charge cost per item in the `cvp_list`. Is each such item its own operation, which should be charged the cost of `AGG_SIG`?

Reply:
Yes it looks like it's set at 20 cost here.
https://github.com/Chia-Network/chia-blockchain/blob/bb06f0b8d49b3525b1d91219615c769896c93d7a/src/consensus/condition_costs.py#L4

There is a 108 multiplier to compare to clvm cost: 
https://github.com/Chia-Network/chia-blockchain/blob/dev/src/consensus/default_constants.py#L54

That would imply a cost of 2100 which is probably a little bit on the cheap side. 

Also, not sure why we have two types of costs, wouldn't it be simpler to use the same units everywhere?

Reply:
My comment above wasn't very clear, so I'll rewrite.

As we know, "running" a block generates a list of coins consumed and their corresponding condition outputs. Most of these conditions can be checked extremely quickly ("does block timestamp exceed X?"), so provide no DOS vectors. But `AGG_SIG` and `AGG_SIG_ME` invoke slow bls operations, so when those conditions are present, we have to add some cost to the block program to ensure creating a zillion of these condition outputs isn't a DOS vector.

Someone should benchmark these bls operations to ensure the cost we charge for `AGG_SIG*` is correctly proportional to all the other costs.